### PR TITLE
Fix extender registration failure with multiple registrations

### LINKED ISSUE #2888: Extender registration failure with multiple registrations
**Describe the bug**

This is a regression, previously fixed.
When multiple extensions register with Zowe Explorer only the first registered is able to be used in trees. The other extension registration will fail after the first registered resulting in the second extension type not being accessible in Zowe Explorer.

**Expected behavior**

All extenders that register with Zowe Explorer don't have to be the first registered to work, all should work and be available.

**Screenshots**

**Desktop (please complete the following information):**

- OS:
- Zowe Explorer Version: 2.15.4
- (Optional) Zowe CLI Version:
- (Optional) Are you using Secure Credential Store?

**Additional context**

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

Fixes #2888. To test this PR, install multiple extensions that contribute profiles to a ZE tree view (like the FTP extension and the USS sample extension). They should fail to load with ZE 2.15.4, but work with a VSIX built from this PR.

## Release Notes

Milestone: 2.16.1?

Changelog: Fixed issue where multiple extensions that contribute profiles to a tree view using the Zowe Explorer API may fail to load

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

Thanks @awharn for helping to write tests 😋

### CODE REVIEW

**Review by JillieBeanSim — COMMENTED** [review]
Thanks @t1m0thyj for tackling this bug, I do see all registered types showing as available.
I do see an issue with the FTP profile. I was prompted for credentials (user/pass in base for all) and zosmf and other prof type (rse used) ds filter search worked but the zftp kept having pop up with invalid creds

**Comment by JillieBeanSim** [comment]
> Thanks @t1m0thyj for tackling this bug, I do see all registered types showing as available. I do see an issue with the FTP profile. I was prompted for credentials (user/pass in base for all) and zosmf and other prof type (rse used) ds filter search worked but the zftp kept having pop up with invalid creds

Sorry after more testing this credential issue exists on marketplace version as well.

**Review by JillieBeanSim — DISMISSED** [review]

File: packages/zowe-explorer/src/dataset/DatasetTree.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ const localize: nls.LocalizeFunc = nls.loadMessageBundle();
 export async function createDatasetTree(log: imperative.Logger): Promise<DatasetTree> {
     const tree = new DatasetTree();
     tree.initializeFavorites(log);
-    await tree.addSession(undefined, undefined, tree);
+    await tree.addSession();
```
Comment (traeok) [inline packages/zowe-explorer/src/dataset/DatasetTree.ts:66]: Thanks for cleaning up this function call 😋

**Review by traeok — DISMISSED** [review]
LGTM - thanks @t1m0thyj

**Comment by JillieBeanSim** [comment]
I think with 2.16.0 about to go out the door, we wait to merge this until after the release.  We can discuss the plan of 2.16.1 and put together a list of fixes we want to include and when we would like it to go out, 1 week? 2 weeks?.

**Review by JillieBeanSim — APPROVED** [review]

**Pull request merged**